### PR TITLE
FLW-866 View Crowdloans

### PR DESCRIPTION
**Pull request by ERussel**
- add crowdloan list screen
Simulator Screen Shot - iPhone 12 - 2021-05-24 at 16 56 36

### CODE REVIEW

File: fearless/Common/Services/StorageQueryService/StorageKeysQueryService.swift [hunk 1]
```diff
@@ -0,0 +1,169 @@
...
+        }
+    }
+
+    private func completeWithResult(_ result: [T]) {
+        let closure = completionClosure
+
+        completionClosure = nil
+        state = .completed(result: result)
+
+        closure?(.success(result))
```
Comment (ef1rspb) [inline fearless/Common/Services/StorageQueryService/StorageKeysQueryService.swift:120]:
```swift
state = .completed(result: result)       
completionClosure?(.success(result))
completionClosure = nil

File: fearless/Common/View/StatusSectionView.swift [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+        setupLayout()
+    }
+
+    @available(*, unavailable)
+    required init?(coder _: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
+    }
+
+    private func setupLayout() {
+        addSubview(indicatorView)
```
Comment (ef1rspb) [inline fearless/Common/View/StatusSectionView.swift:32]: contentView.addSubview(...)

File: fearless/Common/View/StatusSectionView.swift [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+        indicatorView.snp.makeConstraints { make in
+            make.leading.equalToSuperview().inset(UIConstants.horizontalInset)
+            make.centerY.equalToSuperview()
+            make.width.height.equalTo(8.0)
```
Comment (ef1rspb) [inline fearless/Common/View/StatusSectionView.swift:36]: just `.size.equalTo(..)`

File: fearless/Common/ViewModel/RemoteImageViewModel.swift [hunk 4]
```diff
@@ -0,0 +1,57 @@
+import UIKit
+import Kingfisher
+import SVGKit
+
+final class RemoteImageViewModel: NSObject {
+    private var url: URL
```
Comment (ef1rspb) [inline fearless/Common/ViewModel/RemoteImageViewModel.swift:6]: let?

File: fearless/Modules/Crowdloan/CrowdloanList/CrowdloanListViewController.swift [hunk 5]
```diff
@@ -0,0 +1,274 @@
...
+
+        applyState()
+    }
+}
+
+extension CrowdloanListViewController: Localizable {
+    func applyLocalization() {
+        if isViewLoaded {
+            setupLocalization()
+        }
```
Comment (ef1rspb) [inline fearless/Modules/Crowdloan/CrowdloanList/CrowdloanListViewController.swift:226]: tableView.reloadData()?

File: fearless/Modules/Crowdloan/CrowdloanList/View/ActiveCrowdloanTableViewCell.swift [hunk 6]
```diff
@@ -0,0 +1,154 @@
...
+            bottom: 0.0,
+            right: UIConstants.horizontalInset
+        )
+    }
+
+    private func setupLayout() {
+        contentView.addSubview(iconImageView)
+
+        iconImageView.snp.makeConstraints { make in
+            make.left.equalToSuperview().inset(UIConstants.horizontalInset)
```
Comment (ef1rspb) [inline fearless/Modules/Crowdloan/CrowdloanList/View/ActiveCrowdloanTableViewCell.swift:109]: leading?

File: fearless/Modules/Crowdloan/CrowdloanList/View/ActiveCrowdloanTableViewCell.swift [hunk 7]
```diff
@@ -0,0 +1,154 @@
...
+    }()
+
+    private var viewModel: ActiveCrowdloanViewModel?
+
+    override func prepareForReuse() {
+        super.prepareForReuse()
+
+        viewModel?.iconViewModel.cancel(on: iconImageView)
+        iconImageView.image = nil
+    }
```
Comment (ef1rspb) [inline fearless/Modules/Crowdloan/CrowdloanList/View/ActiveCrowdloanTableViewCell.swift:53]: viewModel = nil

File: fearless/Modules/Crowdloan/CrowdloanList/View/ActiveCrowdloanTableViewCell.swift [hunk 8]
```diff
@@ -0,0 +1,154 @@
...
+
+        titleLabel.snp.makeConstraints { make in
+            make.leading.equalTo(iconImageView.snp.trailing).offset(UIConstants.horizontalInset)
+            make.top.equalToSuperview().inset(12.0)
+        }
+
+        contentView.addSubview(navigationImageView)
+
+        navigationImageView.snp.makeConstraints { make in
+            make.right.equalToSuperview().inset(UIConstants.horizontalInset)
```
Comment (ef1rspb) [inline fearless/Modules/Crowdloan/CrowdloanList/View/ActiveCrowdloanTableViewCell.swift:124]: trailing

File: fearless/Modules/Crowdloan/CrowdloanList/View/CompletedCrowdloanTableViewCell.swift [hunk 9]
```diff
@@ -0,0 +1,124 @@
...
+            bottom: 0.0,
+            right: UIConstants.horizontalInset
+        )
+    }
+
+    private func setupLayout() {
+        contentView.addSubview(iconImageView)
+
+        iconImageView.snp.makeConstraints { make in
+            make.left.equalToSuperview().inset(UIConstants.horizontalInset)
```
Comment (ef1rspb) [inline fearless/Modules/Crowdloan/CrowdloanList/View/CompletedCrowdloanTableViewCell.swift:94]: leading

**Review by ef1rspb — APPROVED** [review]

File: fearless/Common/Services/StorageQueryService/StorageKeysQueryService.swift (hunk 1 above)
Comment (ERussel) [reply fearless/Common/Services/StorageQueryService/StorageKeysQueryService.swift:120]: The main purpose here is that we don't have any assumption that the service will not be called in closure

File: fearless/Modules/Crowdloan/CrowdloanList/CrowdloanListViewController.swift (hunk 5 above)
Comment (ERussel) [reply fearless/Modules/Crowdloan/CrowdloanList/CrowdloanListViewController.swift:226]: As you can see I added localization manager also to presenter. I thought it has several benefits: 
1) Presenter decides whether model depends on localization and reloads view if needed, an view just decides localization for a static content;
2) We don't have overhead with ```LocalizableResource``` stuff anymore.

**Pull request merged**